lib/python/pyflyby/_interactive.py

```diff
@@ -759,174 +753,65 @@ def get_global_namespaces(ip):
-        results = sorted([r for r in results if r.startswith(attrname)])
-    elif len(splt) == 2:
-        # Check members, including known sub-modules and importable sub-modules.
-        pname = splt[0]
-        if allow_eval:
-            # Evaluate the parent, with autoimporting.
-            ns_g, ns_l = namespaces.merged_to_two()
-            # TODO: only catch exceptions around the eval, not the other stuff
-            # (loading import db, etc).
-            try:
-                parent = auto_eval(pname, globals=ns_g, locals=ns_l, db=db)
-            except Exception as e:
-                logger.debug("complete_symbol(%r): couldn't evaluate %r: %s: %s",
-                             fullname, pname, type(e).__name__, e)
-                return []
-        else:
-            try:
-                parent = load_symbol(pname, namespaces, autoimport=True, db=db,
-                                     autoimported=autoimported)
-            except LoadSymbolError as e2:
-                # Even after attempting auto-import, the symbol is still
-                # unavailable, or some other error occurred.  Nothing to complete.
-                e3 = getattr(e2, "__cause__", e2)
-                logger.debug("complete_symbol(%r): couldn't load symbol %r: %s: %s",
-                             fullname, pname, type(e3).__name__, e3)
-                return []
-        logger.debug("complete_symbol(%r): %s == %r", fullname, pname, parent)
-        results = set()
-        # Add current attribute members.
-        results.update(_list_members_for_completion(parent, ip))
-        # Is the parent a package/module?
-        if sys.modules.get(pname, Ellipsis) is parent and parent.__name__ == pname:
-            # Add known_imports entries from the database.
-            results.update(known.member_names.get(pname, []))
-            # Get the module handle.  Note that we use ModuleHandle() on the
-            # *name* of the module (``pname``) instead of the module instance
-            # (``parent``).  Using the module instance normally works, but
-            # breaks if the module hackily replaced itself with a pseudo
-            # module (e.g. https://github.com/josiahcarlson/mprop).
-            pmodule = ModuleHandle(pname)
-            # Add importable submodules.
-            results.update([m.name.parts[-1] for m in pmodule.submodules])
-        results = sorted([r for r in results if r.startswith(attrname)])
-        results = ["%s.%s" % (pname, r) for r in results]
-    else:
-        raise AssertionError
-    logger.debug("complete_symbol(%r) => %r", fullname, results)
-    return results
-
+        results = sorted([r for r in results])
+        return results
 
-def _list_members_for_completion(obj, ip):
-    """
-    Enumerate the existing member attributes of an object.
-    This emulates the regular Python/IPython completion items.
+    def keys(self):
+        return list(dict.keys(self)) + self._potential_imports_list
```

Comment:
```suggestion
        return list(self) + self._potential_imports_list
```

Reply:
I think this inherits from `dict`, so `list(<dict>)` should return a list of the keys?